apps/web/frontend/components/weekSelector.tsx

```diff
@@ -1,85 +1,69 @@
-import {
-  eachWeekOfInterval,
-  endOfWeek,
-  endOfYear,
-  format,
-  getWeek,
-  getYear,
-  isSameYear,
-  isThisWeek,
-  setWeek,
-  setYear,
-  startOfWeek,
-  startOfYear,
-} from 'date-fns'
+import { endOfWeek, format, getWeek, getYear, isThisWeek, startOfWeek, nextMonday, previousMonday } from 'date-fns'
+import { enGB } from 'date-fns/locale'
+import { useState } from 'react'
+import { BiLeftArrow, BiRightArrow } from 'react-icons/bi'
+import { BsCalendarCheck } from 'react-icons/bs'
+
+import { CalendarSelector } from './calendarSelector'
 
 interface WeekSelectorProps {
   value: Date
   onChange: (newDay: Date) => void
 }
 
 export const WeekSelector = ({ value, onChange }: WeekSelectorProps) => {
-  const selectedDay = startOfWeek(value, { weekStartsOn: 1 })
-  const selectedYear = getYear(selectedDay)
-  const selectedWeek = getWeek(selectedDay, { weekStartsOn: 1, firstWeekContainsDate: 7 })
-
-  const startOfSelectedYear = startOfYear(selectedDay)
-  const endOfSelectedYear = endOfYear(selectedDay)
-
-  const weeksOfSelectedYear = eachWeekOfInterval(
-    { start: startOfSelectedYear, end: endOfSelectedYear },
-    { weekStartsOn: 1 },
-  ).filter((startOfWeekDate) => isSameYear(startOfWeekDate, selectedDay))
+  const [selectedWeek, setSelectedWeek] = useState({
+    startDate: startOfWeek(value, { locale: enGB }),
+    endDate: endOfWeek(value, { locale: enGB }),
+    year: getYear(value),
+    key: 'selection',
+  })
+  const selectedCalendarWeek = getWeek(selectedWeek.startDate, { locale: enGB })
+  const isCurrentWeek = isThisWeek(selectedWeek.startDate, { weekStartsOn: 1 })
 
-  const thisYear = getYear(new Date())
-  const years = [thisYear, thisYear - 1, thisYear - 2]
-
-  const handleWeekChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    const newWeek = Number.parseInt(event.target.value)
-    const newDay = setWeek(selectedDay, newWeek, { weekStartsOn: 1, firstWeekContainsDate: 7 })
-    onChange(newDay)
+  const getDaysOfWeek = (date: Date) => {
+    const monday = startOfWeek(date, { locale: enGB })
+    const sunday = endOfWeek(date, { locale: enGB })
+    const year = getYear(date)
+    setSelectedWeek({ startDate: monday, endDate: sunday, year, key: 'selection' })
+    onChange(monday)
   }
 
-  const handleYearChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
-    const newYear = Number.parseInt(event.target.value)
-    const newDay = setYear(selectedDay, newYear)
-    onChange(newDay)
+  const handleWeekSelect = (newDate: Date) => {
+    getDaysOfWeek(newDate ?? value)
```

Comment:
`newDate` is never `undefined` or `null`. So what is the point of nullish coalescing here?